sasmodels/models/two_yukawa.py

```diff
@@ -0,0 +1,193 @@
+here may be influenced by the numerical accuracy of a computer. In general, when
+$Z<20$, the code should function well for most cases. When $Z>25$, sometimes the
+intermediate results of this code may run into the limit of the largest number
+that a computer can handle. Therefore, results may potentially become less
+reliable. Hence, the check of $g(r)$ becomes very important in those situations.
+So far, no limitation of the value of $K$ has been encountered except that they
+cannot be zero.
+
+References
+----------
+
+#. Y. Liu, W. R. Chen, S-H Chen, *J. Chem. Phys.*, 122 (2005) 044507
+
+#. M. Broccio, D. Costa, Y. Liu, S-H Chen, *J. Chem. Phys.*, 124 (2006) 084501
+
+#. M. Kotlarchyk and S-H Chen, *J. Chem. Phys.*, 79 (1983) 2461-2469
+
+#. J. S. Hoye, L. Blum, J. Stat. Phys., 16(1977) 399-413
+
+Authorship and Verification
+---------------------------
+
+* **Author:** Yun Liu **Date:** January 22, 2024
+* **Last Modified by:** Yun Liu **Date:** January 22, 2024
+* **Last Reviewed by:** Yun Liu **Date:** January 22, 2024
+"""
+import numpy as np
+from numpy import inf
+
+# TODO: move TwoYukawa library to sasmodels.TwoYukawa and drop the complicated import machinery.
+# two_yukawa requires the TwoYukawa package. This should be available in
+# sasmodels, but if you are using a custom version in the plugins directory
+# you will want to use your own version of the library. The following hack
+# loads TwoYukawa from the same location as two_yukawa
+import sys
+from pathlib import Path
+import importlib.util
+
+# TODO: pep8 says packages and modules should not use camel case
+from sasmodels.TwoYukawa.CalTYSk import CalTYSk, Q_UPPER, Q_STEP, K_MIN, Z_MIN, Z_MIN_DIFF
+
+# If you want a customized version of two_yukawa as a plugin (for example, because you want
+# to use the high precision polynomial root solver from mpmath) it will need to find the
+# customized TwoYukawa library. The following loads the most recent version of the TwoYukawa
+# library from the current directory each time the plugin is leaded. It assumes the TwoYukawa
+# library is in a subdirectory beside the two_yukawa.py plugin model.
+# Note: to test this code, comment the import line above, and change path to path.parent in
+# the spec = ... line below.
+if 0:
+    # Remove existing TwoYukawa from sys.modules to force a reload
+    remove = [modname for modname in sys.modules if modname.startswith('TwoYukawa.') or modname == 'TwoYukawa']
+    for modname in remove:
+        del sys.modules[modname]
+
+    # Load or reload the TwoYukawa package
+    path = Path(__file__).resolve().parent
+    spec = importlib.util.spec_from_file_location('TwoYukawa', str(path / 'TwoYukawa' / '__init__.py'))
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    sys.modules['TwoYukawa'] = module
+
+    # Override sasmodels library symbols with the local symbols.
+    from TwoYukawa.CalTYSk import CalTYSk, Q_UPPER, Q_STEP, K_MIN, Z_MIN, Z_MIN_DIFF
```

Comment:
This entire setup is a bit presumptive that someone will implement a separate calculator and if that happens, the person will use the exact names defined by this implementation.

Reply:
Note that I didn't test the custom plugin method with bumps fitting. It might fail when pickling the model and sending it to a fit process.